openmmtools/multistate/sams.py

```diff
@@ -222,17 +234,23 @@ def __init__(self,
         self.log_target_probabilities = log_target_probabilities
         self.state_update_scheme = state_update_scheme
         self.locality = locality
+        self.weight_update = weight_update
         self.update_stages = update_stages
-        self.flatness_criteria = flatness_criteria
-        self.flatness_threshold = flatness_threshold
         self.weight_update_method = weight_update_method
+        self.beta_factor = beta_factor
         self.adapt_target_probabilities = adapt_target_probabilities
         self.gamma0 = gamma0
         self.logZ_guess = logZ_guess
         # Private variables
         # self._replica_neighbors[replica_index] is a list of states that form the neighborhood of ``replica_index``
         self._replica_neighbors = None
         self._cached_state_histogram = None
+        self.criteria = { criteria: value for criteria, value in (('flatness', flatness),  ('round_trips', round_trips),
+                                                                 ('minimum_logZ', minimum_logZ),  ('minimum_visits', minimum_visits))
+                                                                 if value is not None}
+        self.advance = {criteria: False for criteria, value in self.criteria.items() if value is not None}
```

Comment:
I don't think we need to expose this in the public API so I'd call this `_advance` instead. If we want to allow the users to force the next stage, we can instead expose the `self._stage` attribute.